File: cpp/linked_list.cpp
#include <iostream>

using namespace std;

class Node {
public:
    int data;
    Node* next;
    
    Node(int value) {
        this->data = value;
        this->next = NULL;
    }
};

class LinkedList {
public:
    Node* head;
    
    LinkedList() {
        head = NULL;
    }
    
    void add(int value) {
        if (head == NULL) {
            head = new Node(value);
            return;
        }
        
        Node* node = head;
        while (node->next != NULL) {
            node = node->next;
        }
        node->next = new Node(value);
    }
};

int main() {
    LinkedList nums;
    nums.add(1);
    nums.add(2);
    nums.add(3);
    
    Node* itr = nums.head;
    while (itr) {
        cout << itr->data << endl;
        itr = itr->next;
    }

    return 0;
}

File: javascript/linkedList.js

class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    constructor() {
        this.head = null;
    }
    
    add(value) {
        if (this.head == null) {
            this.head = new Node(value);
            return;
        }
        
        let node = this.head;
        
        while (node.next != null) {
            node = node.next;
        }
        
        node.next = new Node(value);
    }
}

let listNums = new LinkedList();
listNums.add(1);
listNums.add(2);
listNums.add(3);
listNums.add(4);

let node = listNums.head;
while (node != null) {
    console.log(`node: [${node.data}]`);
    node = node.next;
}